pkg/provider/provider.go

```diff
@@ -38,183 +44,173 @@ func init() {
-						return nil, nil
-					default:
-						errs := append(errs, fmt.Errorf("%q must be one of Snowflake, OAuth, ExternalBrowser, Okta, JWT, TokenAccessor or UsernamePasswordMFA", key))
-						return warns, errs
-					}
-				},
+				Type:             schema.TypeString,
+				Description:      envNameFieldDescription("Specifies the [authentication type](https://pkg.go.dev/github.com/snowflakedb/gosnowflake#AuthType) to use when connecting to Snowflake. Valid values include: Snowflake, OAuth, ExternalBrowser, Okta, JWT, TokenAccessor, UsernamePasswordMFA. It has to be set explicitly to JWT for private key authentication.", snowflakeenvs.Authenticator),
+				Optional:         true,
+				DefaultFunc:      schema.EnvDefaultFunc(snowflakeenvs.Authenticator, string(authenticationTypeSnowflake)),
+				ValidateDiagFunc: normalizeValidation(toAuthenticatorType),
 			},
 			"passcode": {
 				Type:          schema.TypeString,
-				Description:   "Specifies the passcode provided by Duo when using multi-factor authentication (MFA) for login. Can also be sourced from the `SNOWFLAKE_PASSCODE` environment variable. ",
+				Description:   envNameFieldDescription("Specifies the passcode provided by Duo when using multi-factor authentication (MFA) for login.", snowflakeenvs.Passcode),
 				Optional:      true,
 				ConflictsWith: []string{"passcode_in_password"},
-				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PASSCODE", nil),
+				DefaultFunc:   schema.EnvDefaultFunc(snowflakeenvs.Passcode, nil),
 			},
 			"passcode_in_password": {
 				Type:          schema.TypeBool,
-				Description:   "False by default. Set to true if the MFA passcode is embedded in the login password. Appends the MFA passcode to the end of the password. Can also be sourced from the `SNOWFLAKE_PASSCODE_IN_PASSWORD` environment variable. ",
+				Description:   envNameFieldDescription("False by default. Set to true if the MFA passcode is embedded to the configured password.", snowflakeenvs.PasscodeInPassword),
 				Optional:      true,
 				ConflictsWith: []string{"passcode"},
-				DefaultFunc:   schema.EnvDefaultFunc("SNOWFLAKE_PASSCODE_IN_PASSWORD", nil),
+				DefaultFunc:   schema.EnvDefaultFunc(snowflakeenvs.PasscodeInPassword, nil),
 			},
 			"okta_url": {
-				Type:        schema.TypeString,
-				Description: "The URL of the Okta server. e.g. https://example.okta.com. Can also be sourced from the `SNOWFLAKE_OKTA_URL` environment variable.",
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_OKTA_URL", nil),
+				Type:             schema.TypeString,
+				Description:      envNameFieldDescription("The URL of the Okta server. e.g. https://example.okta.com.", snowflakeenvs.OktaUrl),
+				Optional:         true,
+				DefaultFunc:      schema.EnvDefaultFunc(snowflakeenvs.OktaUrl, nil),
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsURLWithHTTPorHTTPS),
```

Comment:
Yeah, we use Capital names instead of UPPER case, but this function is from the plugin SDK ([docs](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/validation#IsURLWithHTTPorHTTPS)).